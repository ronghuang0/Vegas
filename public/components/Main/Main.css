.tabs {
  width: 100%;
}

.tabList {
  background-color: #875F94;
  color: #cccccc;
  display: table;
  width: 100%;
  padding: 0;
  margin: 0;
}

.tab {
  display: table-cell;
  list-style: none;
  padding: 18px 12px;
  cursor: pointer;
  text-align: center;
}

.tab:hover {
  border-bottom: 3px solid #EB8631;
  color: white;
}
.tabList .tabSelected {
  border-bottom: 3px solid indigo;
  color: white;
}

.tabPanel {
  margin: 15px;
}

.map {
  width: 100%;
}

.detailsTitle {
  font-weight: bold;
}

.detailsText {
  margin-bottom: 12px;
}
