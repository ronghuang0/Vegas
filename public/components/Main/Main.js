import React, { useState } from 'react';
import {
  Tab,
  Tabs,
  TabList,
  TabPanel,
} from 'react-tabs';
import Expandable from '../Expandable/Expandable';
import Header from '../Header/Header';
import styles from './Main.css';

const Main = ({ data }) => {
  const [selectedTab, setSelectedTab] = useState(0);
  const {
    description,
    details,
    media = [],
    name,
    phoneNumber,
    location,
    price,
    starRating,
  } = data;
  const productMap = media.find((element) => element.type === 'productMap');
  const handleSelect = (tabIndex) => setSelectedTab(tabIndex);
  return (
    Object.keys(data).length !== 0 && (
      <>
        <Header
          name={name}
          phoneNumber={phoneNumber}
          location={location}
          price={price}
          starRating={starRating}
          onSelect={handleSelect}
        />
        <Tabs className={styles.tabs} selectedIndex={selectedTab} onSelect={handleSelect}>
          <TabList className={styles.tabList}>
            <Tab selectedClassName={styles.tabSelected} className={styles.tab}>DESCRIPTION</Tab>
            <Tab selectedClassName={styles.tabSelected} className={styles.tab}>DETAILS</Tab>
            <Tab selectedClassName={styles.tabSelected} className={styles.tab}>LOCATION</Tab>
          </TabList>

          <TabPanel className={styles.tabPanel}>
            <Expandable collapsedText='SHOW FULL DESCRIPTION' expandedText='HIDE FULL DESCRIPTION'>
              {
                description.split('\r\n').map((substring) => (
                  <>
                    {substring}
                    <br />
                  </>
                ))
              }
            </Expandable>
          </TabPanel>
          <TabPanel className={styles.tabPanel}>
            <Expandable collapsedText='VIEW MORE DETAILS' expandedText='VIEW FEWER DETAILS'>
              { details.map((detail) => (
                <>
                  <span className={styles.detailsTitle}>{`${detail.label}:`}</span>
                  <div className={styles.detailsText}>{detail.value}</div>
                </>
              ))}
            </Expandable>
          </TabPanel>
          <TabPanel className={styles.tabPanel}>
            { productMap && <img className={styles.map} src={productMap.href} /> }
          </TabPanel>
        </Tabs>
      </>
    ));
};

export default Main;
