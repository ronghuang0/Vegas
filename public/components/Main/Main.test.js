import React from 'react';
import renderer from 'react-test-renderer';
import Main from './Main';

describe('Main', () => {
  it('should render correctly if data is loaded', () => {
    const data = {
      description: 'description',
      details: [{}],
      media: [],
      name: 'name',
      phoneNumber: 'phoneNumber',
      location: {},
      price: 'price',
      starRating: 'starRating',
    };
    const component = renderer.create(<Main data={data} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not render if data is not loaded', () => {
    const component = renderer.create(<Main data={{}} />);
    const tree = component.toJSON();
    expect(tree).toEqual(null);
  });
});
