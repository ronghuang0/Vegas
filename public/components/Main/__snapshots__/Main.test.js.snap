// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly if data is loaded 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="leftHeader"
    >
      <div
        className="title"
      >
        <div
          className="name"
        >
          name
        </div>
        <div
          className="halfStarIcon"
        />
      </div>
      <div
        className="info"
      >
        <div
          className="location"
          onClick={[Function]}
        >
          <div
            className="markIcon"
          />
        </div>
        <div>
          <div
            className="phoneIcon"
          />
          phoneNumber
        </div>
        <div>
          <div
            className="likeIcon"
          />
          Best Price Guarantee
        </div>
      </div>
    </div>
    <div
      className="rightHeader"
    >
      <div
        className="price"
      >
        $price
      </div>
      <div
        className="priceText"
      >
        HOTEL ROOMS FROM
      </div>
    </div>
  </div>,
  <div
    className="tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="tabList"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        className="tab tabSelected"
        id="react-tabs-0"
        role="tab"
        tabIndex="0"
      >
        DESCRIPTION
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        className="tab"
        id="react-tabs-2"
        role="tab"
        tabIndex={null}
      >
        DETAILS
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        className="tab"
        id="react-tabs-4"
        role="tab"
        tabIndex={null}
      >
        LOCATION
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      className="tabPanel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      <div
        className="collapsed"
      >
        description
        <br />
      </div>
      <div
        className="expandText"
        onClick={[Function]}
      >
        SHOW FULL DESCRIPTION
        <div
          className="downIcon"
        />
      </div>
    </div>
    <div
      aria-labelledby="react-tabs-2"
      className="tabPanel"
      id="react-tabs-3"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-4"
      className="tabPanel"
      id="react-tabs-5"
      role="tabpanel"
    />
  </div>,
]
`;
